executor/adapter.go

```diff
@@ -826,6 +828,17 @@ func getPlanTree(p plannercore.Plan) string {
 	return variable.SlowLogPlanPrefix + planTree + variable.SlowLogPlanSuffix
 }
 
+// getPlanTree will try to get the select plan tree if the plan is select or the select plan of delete/update/insert statement.
+func getPlanDigest(sctx sessionctx.Context, p plannercore.Plan) (normalized, planDigest string) {
+	normalized, planDigest = sctx.GetSessionVars().StmtCtx.GetPlanDigest()
+	if len(normalized) > 0 {
+		return
+	}
+	normalized, planDigest = plannercore.NormalizePlan(p)
```

Comment:
1. Can you move this logic into `StmtCtx.GetPlanDigest()`?
2. Is it possible that normalized is always empty for some kind of SQL, like DAL/DDL? So `len(normalized) > 0` is always false.

Reply:
1. no, it will have cycle import.
2. yes, it is possible, then it will do `plannercore.NormalizePlan(p)` again, but it is fast since the result is `''`.